(chore) O3-1109: Added a GitHub Action to check PR titles

IamLRBA opened the pull request:
Duplicate PR of https://github.com/openmrs/openmrs-esm-core/pull/1307#event-16962739290

# Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. Ensure your PR title includes a [conventional commit](https://o3-docs.openmrs.org/docs/frontend-modules/contributing#contributing-guidelines) label (such as `feat`, `fix`, or `chore`, among others). See existing PR titles for inspiration.

## For changes to apps

- [x] My work conforms to the [**O3 Styleguide**](https://om.rs/styleguide) and [**design documentation**](https://om.rs/o3ui).

## If applicable

- [x] My work includes tests or is validated by existing tests.
- [ ] I have updated the [esm-framework mock](https://github.com/openmrs/openmrs-esm-core/blob/main/packages/framework/esm-framework/mock.tsx) to reflect any API changes I have made.

## Summary
This PR adds a GitHub Actions workflow to enforce PR title conventions using the `amannn/action-semantic-pull-request@v5` action. The workflow checks whether PR titles follow the conventional commit format (`feat`, `fix`, `docs`, or `BREAKING`:). This ensures consistency in PR naming, improving clarity in commit history and changelogs.  

## Screenshots
N/A (This is a CI/CD workflow update).  

## Related Issue
Related Jira [issue](https://openmrs.atlassian.net/jira/software/c/projects/O3/issues/O3-1109?jql=project%20%3D%20%22O3%22%20AND%20status%20%3D%20%22To%20Do%22%20AND%20assignee%20%3D%20empty%20AND%20%22complexity%5Bdropdown%5D%22%20%3D%20Low%0AORDER%20BY%20created%20DESC)

## Other
This automation will help maintain standardized PR titles across the repository, reducing manual review overhead.  



<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

- **Chores**
	- Introduced an automated check to ensure pull request titles follow a standardized format, enhancing consistency in contributions.

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

IamLRBA commented:
@brandones I created this fresh PR on the same issue because the previous had branch conflicts on my local repository. Also there were too many commits on the previous one. I thought one commit having all the changes thus far would be better.

brandones commented:
Part of the reason why we don't want PRs closed and replaced with new ones is because we lose the comment and revision history. And you have not fixed the problem I named last time. The check needs to pass with the usual PR title format that the OpenMRS community uses.

[My previous comment](https://github.com/openmrs/openmrs-esm-core/pull/1307#issuecomment-2722874265).

IamLRBA commented:
> Do not open new PRs for the same work if you can possibly avoid it. I understand getting into messes with git branches. Try to learn to use git better. AI can be useful for learning git and debugging messed up git state.
> 
> 
> 
> Too many commits is not a problem. They get squashed when a PR is merged anyway. If you have commits locally that you want to squash before you push, you can use an interactive rebase to do that.

Thank you for the advice @brandones I really appreciate it and the chance to learn.
These mistakes will not exist the next time.

IamLRBA commented:
> Part of the reason why we don't want PRs closed and replaced with new ones is because we lose the comment and revision history. And you have not fixed the problem I named last time. The check needs to pass with the usual PR title format that the OpenMRS community uses.
> 
> 
> 
> [My previous comment](https://github.com/openmrs/openmrs-esm-core/pull/1307#issuecomment-2722874265).
> 
> 
> 
> 

The previous PR with the title "(chore) O3-1109: Added a GitHub Action to Check PR titles
(chore) Added a GitHub Action to Check PR titles" had the PR title test failing and when I did some research I discovered the "()" brackets made it fail so I neglected that and this current PR has that particular test passing.

brandones commented:
@IamLRBA  Do you understand what the problem is and what I am asking you to do?

brandones commented:
Anyone who wants to carry this across the finish line should feel free. It's an easy fix and would be cool to have this feature. Here is what needs to happen:

Use the headerPattern and headerPatternCorrespondence options of action-semantic-pull-request so that PR titles matching our norm pass. These titles should pass:

```
(chore) O3-1109: Added a GitHub Action to Check PR titles
(chore) Added a GitHub Action to Check PR titles
```

These titles should fail:

```
O3-1109: Added a GitHub Action to Check PR titles
Added a GitHub Action to Check PR titles
(chore) 03-1109: Added a GitHub Action to Check PR titles
```

That last one uses zero-three in place of O3.
